libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -246,6 +252,14 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
         verify(exactly = 1) { tripSession.unregisterAllRoadObjectsOnRouteObservers() }
     }
 
+    @Test
+    fun onDestroy_unregisters_HistoryRecordingStateHandler_observers() {
+        createMapboxNavigation()
+        mapboxNavigation.onDestroy()
+
+        verify(exactly = 1) { historyRecordingStateHandler.unregisterAllStateChangeObservers() }
+    }
```

Comment:
NIT Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert

```kotlin
@Test
fun onDestroy_unregisters_HistoryRecordingStateHandler_observers() {
    // Arrange
    
    // Act
    
    // Assert
}
```

Note the empty lines between Arrange / Act and Act / Assert.

The same thing applies to some other tests below 👇